### PR TITLE
Update configurations and parsers

### LINKED ISSUE #1101: Add support for GAOMON S630
# Tablet Information

| Name            | Value |
| --------------- | ----- |
| Manufacturer    | GAOMON
| Product Name    | GAOMON S630
| Width           | 152.4mm
| Height          | 129.2mm
| Pressure Levels | 8192
| LPI             | 5080

### LINKED ISSUE #1282: Add support for XP-PEN Deco Fun L
# Tablet Information

| Name            | Value |
| -------- | ----- |
| Manufacturer    | XP-PEN
| Product Name    | XP-PEN Deco Fun L
| Width           | 254mm
| Height          | 159mm
| Pressure Levels | 8.192
| LPI             | 5080

## Diagnostic Information

[Diagnostic.zip](https://github.com/OpenTabletDriver/OpenTabletDriver/files/6874542/Diagnostic.zip)

### LINKED ISSUE #1338: Support Request: KAMVAS 22 PLUS
# Support Request

## Diagnostic Information

image
image

### LINKED ISSUE #1401: Valid inputs being dropped - Hover distance and area issues
## Description

Many valid inputs are being ignored leading certain tablets to have their input cut off before it should be. This can lead to lowered hover distance, and certain parts of areas (specifically some parts of the area deadzones on wacom tablets) being unreachable or having input cut off too soon.
## System Information:

| Name             | Value |
| ---------------- | ----- |
| OpenTabletDriver Version | 0.6.0 Pre-release
| Tablet                   | Tested on CTL-480 but many others likely affected

### LINKED ISSUE #1435: Add support for WoodPad 7 PF0730
# Tablet Information

| Name            | Value |
| --------------- | ----- |
| Manufacturer   | ViewSonic
| Product Name    | WoodPad 7 PF0730
| Width           | 162
| Height          | 101
| Pressure Levels | 4096
| LPI             | 5080

## Diagnostic Information
[diagnostics.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7162693/diagnostics.txt)

### LINKED ISSUE #1448: Gaomon S56K ReportParser problem
## Description
idk if it's my specific gaomon s56k problem or not but
https://github.com/OpenTabletDriver/OpenTabletDriver/blob/04adf913cdaa282ce97d86f0e8ebee12264ef17e/OpenTabletDriver/Vendors/UCLogic/UCLogicReportParser.cs#L6-L16
[tablet-data.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7191270/tablet-data.txt)

That bit is only 00 upon entering tablet "interactive area"
And most of the time it's not
``` json
{
      "VendorID": 9580,
      "ProductID": 110,
      "InputReportLength": 8,
      "OutputReportLength": null,
      "ReportParser": "OpenTabletDriver.Tablet.TabletReportParser",
      "FeatureInitReport": null,
      "OutputInitReport": null,
      "DeviceStrings": {
        "201": "HUION_T156_\\d{6}$"
      },
      "InitializationStrings": [
        200
      ]
}
```
Works fine if i set TabletReportParser manually in the s56k config
[Vid](https://puu.sh/IcqOF/9b0a219147.mp4)
[diagnostics.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7191632/diagnostics.txt)
## System Information:

| Name             | Value |
| ---------------- | ----- |
| Operating System         Arch linux|
| OpenTabletDriver Version OpenTabletDriver v0.5.3.1-r460.g04adf913 |
| Tablet                   Gaomon S56K |

### LINKED ISSUE #1500: Add support for Genius Pen 560
# Tablet Information

| Name            | Value |
| --------------- | ----- |
| Manufacturer    | Genius
| Product Name    | G-Pen 560
| Width           | 152
| Height          | 114
| Pressure Levels | 1024
| LPI             | 2000

## Diagnostic Information
 

[diagnostic.zip](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7322970/diagnostic.zip)

### LINKED ISSUE #1545: Huion h430p not detecting
## Description
Huion H430P tablet not detecting on manjaro

## System Information:

| Name             | Value |
| ---------------- | ----- |
| Operating System         | Linux Manjaro KDE, kernel: 5.14.10-1-MANJARO  |
| OpenTabletDriver Version | 0.5.3.3 |
| Tablet                   | Huion h430p |

Diagnostics:
[otdd.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7402589/otdd.txt)

### PR DESCRIPTION
**Pull request by X9VoiD**
- Closes #1101
- Closes #1282
- Closes #1338
- Closes #1401
- Closes #1435
- Closes #1448
- Closes #1500
- Closes #1545

### CODE REVIEW

File: OpenTabletDriver.Configurations/Configurations/Parblo/A610 Pro.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "Touch": null
+  },
+  "DigitizerIdentifiers": [
+    {
+      "VendorID": 10429,
+      "ProductID": 6403,
+      "InputReportLength": 10,
+      "OutputReportLength": 10,
+      "ReportParser": "OpenTabletDriver.Configurations.Parsers.XP_Pen.XP_PenReportParser",
+      "FeatureInitReport": null,
```
Comment (gonX) [inline OpenTabletDriver.Configurations/Configurations/Parblo/A610 Pro.json:29]:
```suggestion
      "FeatureInitReport": null,
```
cosmetic

File: OpenTabletDriver.Configurations/Parsers/Genius/GeniusTabletReport.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            {
+                X = Unsafe.ReadUnaligned<ushort>(ref report[1]),
+                Y = Unsafe.ReadUnaligned<ushort>(ref report[3])
+            };
+            Pressure = report[5].IsBitSet(2) ? Unsafe.ReadUnaligned<ushort>(ref report[6]) : 0u;
+
+            PenButtons = new bool[]
+            {
+                    report[5].IsBitSet(3),
+                    report[5].IsBitSet(4)
```
Comment (gonX) [inline OpenTabletDriver.Configurations/Parsers/Genius/GeniusTabletReport.cs:23]:
```suggestion
               report[5].IsBitSet(3),
               report[5].IsBitSet(4)
```
one tab too much, might want to just autoindent this file

**Review by gonX — APPROVED** [review]

**Review by gonX — APPROVED** [review]

File: OpenTabletDriver.Configurations/Configurations/ViewSonic/WoodPad PF0730.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+  "Name": "Woodpad PF0730",
```
Comment (InfinityGhost) [inline OpenTabletDriver.Configurations/Configurations/ViewSonic/WoodPad PF0730.json:2]:
```suggestion
  "Name": "ViewSonic Woodpad PF0730",
```

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 22HD.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+  "Name": "Xp-Pen Artist 22HD",
```
Comment (InfinityGhost) [inline OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 22HD.json:2]:
```suggestion
  "Name": "XP-Pen Artist 22HD",
```

File: OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco mini 4.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{
+  "Name": "XP-Pen Deco Mini 4",
```
Comment (InfinityGhost) [inline OpenTabletDriver.Configurations/Configurations/XP-Pen/Deco mini 4.json:2]: Filename does not match name property

**Review by InfinityGhost — CHANGES_REQUESTED** [review]
Formatting is wrong in the following configurations:
- Acepen AP 1060
- Gaomon M10K
- Genius G-Pen 560
- Genius i608x
- Huion 420
- Huion HS64
- Huion Kamvas 16
- Parblo A610
- Wacom ET-0405-U
- XP-Pen Deco 03
- XP-Pen Innovator 16
- XP-Pen Star 03
- XP-Pen Star G430

After required changes are applied, the entire configuration list should be run through the configuration editor to enforce formatting.

**Comment by X9VoiD** [comment]
Should get CI or console lint.

**Comment by InfinityGhost** [comment]
> Should get CI or console lint.

Transferring this to an issue

**Comment by gonX** [comment]
Closes #1448 when merged, since b74dbce3 was added

**Comment by gonX** [comment]
Will also close #1101 because of f111a2a5858

**Comment by gonX** [comment]
..and #1282 because of e4a08ee856d2

**Comment by gonX** [comment]
Some more for the list:

Closes #1545 because of 6c75c3512a
Closes #1338 because of 4bbddbc
Closes #1435 because of 6112c53
Closes #1500 because of ffea0c1

**Comment by gonX** [comment]
Now also closes #1401 because of 5184d421631

File: OpenTabletDriver.Configurations/ReportParserProvider.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using OpenTabletDriver.Plugin;
+using OpenTabletDriver.Plugin.Components;
+using OpenTabletDriver.Plugin.Tablet;
+
+namespace OpenTabletDriver.ComponentProviders
```
Comment (InfinityGhost) [inline OpenTabletDriver.Configurations/ReportParserProvider.cs:9]: This file is left over, or hasn't been moved to the correct path.

**Review by InfinityGhost — CHANGES_REQUESTED** [review]

**Review by InfinityGhost — APPROVED** [review]

**Pull request merged**